File: core/designSystem/src/main/kotlin/com/danielyan/gopeak/designsystem/theme/GoPeakShapes.kt
package com.danielyan.gopeak.designsystem.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Immutable
import androidx.compose.ui.graphics.Shape
import androidx.compose.ui.unit.dp

/** GoPeak corner-radius scale (from Figma: 4 / 8 / 12 / 16). Access via [GoPeakTheme.shapes]. */
@Immutable
data class GoPeakShapes(
    val extraSmall: Shape = RoundedCornerShape(4.dp),
    val small: Shape = RoundedCornerShape(8.dp),
    val medium: Shape = RoundedCornerShape(12.dp),
    val large: Shape = RoundedCornerShape(16.dp),
)

val GoPeakDefaultShapes = GoPeakShapes()

File: iosApp/DesignSystem/Sources/GoPeakDesignSystem/Theme/GoPeakShapes.swift
import SwiftUI

/// GoPeak corner-radius scale (Figma: 4 / 8 / 12 / 16) — counterpart of Android's `GoPeakShapes`.
public struct GoPeakShapes: Sendable {
    public let extraSmall: CGFloat = 4
    public let small: CGFloat = 8
    public let medium: CGFloat = 12
    public let large: CGFloat = 16

    public init() {}

    /// Convenience `RoundedRectangle` for a given radius token.
    public func rounded(_ radius: CGFloat) -> RoundedRectangle {
        RoundedRectangle(cornerRadius: radius, style: .continuous)
    }
}
